### ISSUE TITLE
Prevent Tab Press from popping to Initial Scene

### ISSUE THREAD

**Issue by adriandiaz** [aksonov/react-native-router-flux#1703]
### Version
Tell us which versions you are using: 

- react-native-router-flux v3.38.0
- react-native v0.42.0

I'm trying to familiarize myself with this library, and I have a sample app with two tabs: Tab1 and Tab2. There are two scenes nested in the Tab1 group: scene1 and scene2, where scene1 is the initial view. If I navigate to scene2, then switch tabs by pressing Tab2, then go back to Tab1, it automatically pops scene2 from Tab1 and goes back to scene1 instead of staying on scene2, which is not the behavior I want.

So my question: is it possible to can prevent the views popping to the root when I press the tab bar item? Ideally it would only pop to the root view when I'm already viewing a view within the same tab bar item. So in the example above, when switching from Tab2 to Tab1, when I press Tab1 the first time it stays on scene2, but if I press it again then it pops back to scene1.

Any help/suggestions greatly appreciated!!!

`<Router>
      <Scene key="tabbar" tabs initial tabBarStyle={{ backgroundColor: '#f6f6f6' }}>
        <Scene key="canes_tab" title="Canes" icon={TabIcon}>
          <Scene key="canes_green" component={GreenScene} initial title='Canes Green' />
          <Scene key="canes_orange" component={OrangeScene} title='Canes Orange' />
        </Scene>

        <Scene key="dolphins_tab" title="Dolphins" icon={TabIcon}>
          <Scene key="aqua" component={AquaScene} initial title='Dolphins Aqua' />
          <Scene key="dolphins_orange" component={DolOrangeScene} title='Dolphines Orange' />
        </Scene>
      </Scene>
</Router>`

**Comment by adriandiaz** [comment]
If anyone else is experiencing this issue, just tested it in on the previous version `3.37.0` and that fixed the issue (I was using `3.38.0`. However, `3.37.0` has the issue mentioned on #717 where tapping a tab while you're viewing it doesn't pop back to default scene. So it looks like by attempting to resolve #717 in version `3.38.0` it caused the above issue. Would be sweet if next version could address both cases.

**Comment by esutton** [comment]
@adriandiaz Thank you!

Rolling back to 3.37.0 fixed problem with switching tabs popping back to the starting tab scene!

3.38.1 caused many duplicate dependency errors in the packager
3.38.0 Fixed duplicate dependency errors in the packager

```
    "react": "~15.4.1",
    "react-native": "0.42.3",
    "react-native-router-flux": "3.37.0",
```
